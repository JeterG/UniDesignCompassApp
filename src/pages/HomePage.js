import React from "react";
import {navigate} from "gatsby"
import inter from "../images/home-image-min.jpg";
import obj from "../images/group-working-together.jpg";
import feat from "../images/eie-designprocessmodel-elem.png";
import Layout from '../components/layout';
import { CardDeck, Card, Button } from 'react-bootstrap';

const HomePage = (props) => (
  <Layout>
    <div className="presentation">
      <img id="homeImg" role="presentation" alt={"inter"} src={inter}></img>
      <h1 className="presentation-description">DESIGN COMPASS</h1>
      <h3 className="presentation-description1">to facililate</h3>
      <h1 className="presentation-description2">Creation</h1>

      <Button className="presentation-button" variant="info" size="lg" onClick={() => {navigate("/Login")}}>Begin</Button>
    </div>
    <CardDeck>
      <Card bg="danger" text="white">
        <Card.Header>Objective</Card.Header>
        <Card.Body>
          <Card.Text>
            The Design Compass is a classroom tool for helping students and professionals record and reflect on their design process
            as they work on and complete a design challenge. The Design Compass software provides an interface where students can
            identify and record the various design steps they used while performing them, and add digital notes and pictures to document
            their work. In the analysis view, a record of the work done as well as a graph of time spent on each design step can be shared
            and discussed with instructors or coworkers.
          </Card.Text>
        </Card.Body>
        <Card.Img variant="top" src={obj} style={{paddingLeft: "1em", paddingRight: "1em"}} />
      </Card>
      <Card bg="warning" text="dark">
        <Card.Header>Features</Card.Header>
        <Card.Body>
          <Card.Text>
            The user interface has two pages. The Compass page gives the option of a standard engineering design process model as well as
            a custom model with the means for entering short text-based descriptions by users, and clocks to note time spent on each design
            step. The Log page provides a spreadsheet-styled chronological list of steps taken, and is the place where users can view, edit
            and append the logged information. By clicking on another button in the Log, users can view aggregated data in histogram form
            on steps they have taken during a single design session or an entire design project.
          </Card.Text>
        </Card.Body>
        <Card.Img variant="top" src={feat} style={{paddingLeft: "1em", paddingRight: "1em"}}/>
      </Card>
    </CardDeck>
  </Layout>
);
export default HomePage;
