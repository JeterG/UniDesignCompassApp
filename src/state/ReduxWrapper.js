import React from 'react';
import { Provider } from 'react-redux';
import createStore from "./store";
import { PersistGate } from 'redux-persist/integration/react'

export default ({ element }) => (
  <Provider store={createStore().store}>
    <PersistGate loading={null} persistor={createStore().persistor}>
      {element}
    </PersistGate>
  </Provider>
);