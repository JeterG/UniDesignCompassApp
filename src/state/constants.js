export const AUTH_USER = "AUTH_USER";
export const UPDATE_USER = "UPDATE_USER";
export const LOGOUT_USER = "LOGOUT_USER";
