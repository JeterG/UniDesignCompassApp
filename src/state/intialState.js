export const initialState = {
  user : {},
  isAuthenticated: false,
}