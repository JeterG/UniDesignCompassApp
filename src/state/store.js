import {  createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import RootReducer from './RootReducer';

const composeEnhancers = (typeof window !== `undefined`) ? (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose ) : compose;

export default function configureStore() {
  return createStore(
    RootReducer,
    composeEnhancers(applyMiddleware(thunk))
  );
 }