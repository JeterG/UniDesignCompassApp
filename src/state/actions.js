import { FETCH_USER_INFO, AUTH_USER } from "./constants"

export const fetchUserInfo = (orderID) => ({
  type: FETCH_USER_INFO,
  payload: orderID
})

export const authenticateUser = (authenticated = false) => dispatch => {

  dispatch(
    {
      type: AUTH_USER,
      payload: "true",
      aws_user: [1]
    }
  )
}
