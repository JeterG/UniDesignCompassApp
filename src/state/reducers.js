import {initialState} from "./intialState"
import {AUTH_USER} from "./constants"

export default (state = initialState, action) => {
  switch(action.type) {
    case AUTH_USER :
      return {
        ...state,
        user: action.payload,
        isAuthenticated: true
      };
    default:
      return state;
  }
}

