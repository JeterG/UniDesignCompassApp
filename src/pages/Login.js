import React, { Component } from "react";
import Layout from "../components/layout"
import "../components/bootstrap.css"
import { connect } from 'react-redux';
import { authenticateUser } from '../state/actions';
import { Auth } from "aws-amplify";
import config from "../aws-exports";
import NewPassword from "../components/newPassword";
import SignIn from "../components/SignIn";
import SignUp from "../components/SignUp";
Auth.configure(config);

class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: "",
            password: "",
            repeat_pass:"",
            phone:"",
            username:"",
            user: [],
        };
        this.handleAuth = this.handleAuth.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this._comp = null;
        this.Log_state = "SignIn"
    }

    handleChange = (e)=>{
        this.setState({[e.target.id]:e.target.value});
    }
    handleAuth = (e) => {
        e.preventDefault();
        Auth.signIn(this.state.email, this.state.password)
            .then((res) => {
                switch (res.challengeName) {
                    case "NEW_PASSWORD_REQUIRED":
                        this.Log_state = "NEWPASS";
                        this.setState({
                            user:res
                        })
                        break;
                    case "SOFTWARE_TOKEN_MFA":
                        this.Log_state = "OTP";
                        break;
                    case "MFA_SETUP":
                        break;
                    case "SMS_MFA":
                        break;
                    default:
                        console.log(res);
                        this.Log_state = "LoggedIn";
                        this.props.authenticateUser("true");
                }
                this.forceUpdate();

            },(error)=>{
                switch(error.code){
                    case "UserNotFoundException":
                        alert("User not found Make an account");
                        break;
                    default:

                }
            });
    };
    handlePass=(e)=>{
        e.preventDefault();
        console.log(this.state.user)
        Auth.completeNewPassword(this.state.user,this.state.password)
        .then((res)=>{
            console.log(res);
        },(error)=>{
            console.log(error)
        })
    }
    createAccount=(e)=>{
        e.preventDefault();
        this.Log_state="SIGNUP";
        this.forceUpdate();
    }
    handleCreate=(e)=>{
        e.preventDefault();
        let attributes={username:this.state.username,password:this.state.password,attributes:{email:this.state.email,phone_number:this.state.phone}};
        console.log(attributes);
        Auth.signUp(attributes)
        .then((res)=>{
            this.Log_state="Verify";
        },(error)=>{
            console.log(error);
        })
    }
    determineRender() {
        console.log(this.Log_state);
        switch (this.Log_state) {
            case 'SignIn':
                this._comp = <SignIn handleAuth={this.handleAuth} handleChange={this.handleChange} handleAccount={this.createAccount}/>
                break;
            case 'NEWPASS':
                this._comp=<NewPassword handlePass={this.handlePass} handleChange={this.handleChange}/>
                break;
            case 'MFA_SETUP':
                break;
            case 'OTP':
                break;
            case 'SIGNUP':
                this._comp =<SignUp handleChange={this.handleChange} handleCreate={this.handleCreate} />
                break;
            default:
        }
    }
        render() {
            this.determineRender()
            return (<Layout>{
                this._comp
            }</Layout>
            )
        }
    }
    
function mapStateToProps(state) {
    return {
        isAuthenticated: state.Reducer.isAuthenticated,
    }
}
const mapDispatchToProps = dispatch => ({
    authenticateUser: (auth) => dispatch(authenticateUser(auth))
})
export default connect(mapStateToProps, mapDispatchToProps)(Login);
