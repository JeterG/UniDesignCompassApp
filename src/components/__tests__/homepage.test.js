import React from "react"
import renderer from "react-test-renderer"
import Login from "../../pages/Login";
import { store } from "../../state/store";
import { Provider } from 'react-redux';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import HomePage from '../../pages/HomePage';
configure({adapter:new Adapter()});


const tree = shallow(<HomePage/>);


describe("Testing Home Page",()=>{
    it("Check Render",()=>{
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    it("Check main div",()=>{
        expect(tree.find('div')).toHaveLength(1);
    });
    it("Check Objective and Features" ,()=>{
        expect(tree.find('Card')).toHaveLength(2);
    });
    it("Check Begin Button" ,()=>{
        expect(tree.find('Button.presentation-button')).toHaveLength(1);
    });

})