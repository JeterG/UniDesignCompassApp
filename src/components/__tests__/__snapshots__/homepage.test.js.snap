// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Page Check Render 1`] = `
<Layout>
  <div
    className="presentation"
  >
    <img
      alt="inter"
      id="homeImg"
      role="presentation"
      src="test-file-stub"
    />
    <h1
      className="presentation-description"
    >
      DESIGN COMPASS
    </h1>
    <h3
      className="presentation-description1"
    >
      to facililate
    </h3>
    <h1
      className="presentation-description2"
    >
      Creation
    </h1>
    <Button
      active={false}
      className="presentation-button"
      disabled={false}
      onClick={[Function]}
      size="lg"
      type="button"
      variant="info"
    >
      Begin
    </Button>
  </div>
  <CardDeck>
    <Card
      as="div"
      bg="danger"
      body={false}
      text="white"
    >
      <CardHeader>
        Objective
      </CardHeader>
      <CardBody>
        <CardText>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </CardText>
      </CardBody>
    </Card>
    <Card
      as="div"
      bg="warning"
      body={false}
      text="dark"
    >
      <CardHeader>
        Features
      </CardHeader>
      <CardBody>
        <CardText>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </CardText>
      </CardBody>
    </Card>
  </CardDeck>
</Layout>
`;
